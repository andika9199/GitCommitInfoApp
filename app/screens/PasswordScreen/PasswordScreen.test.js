import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ComponentConnect, {Password, mapDispatchToProps} from './index';

const mockStore = configureMockStore();
const initialProps = {
  login: {
    data: null,
    error: false,
    isFetching: false,
  },
  reqLogin: jest.fn(),
  route: {
    params: {
      username: 'andika9199',
    },
  },
  navigation: {
    reset: jest.fn(),
  },
};
describe('Password Screen', () => {
  describe('Password Screen rendered', () => {
    it('Screen Renderer Successfully', async () => {
      const component = renderer.create(<Password {...initialProps} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Password Screen function', () => {
    const component = shallow(<Password {...initialProps} />);
    const instance = component.instance();

    it('setPassword', () => {
      instance.setPassword('123456');
      expect(instance.state.password).toEqual('123456');
    });
    it('login', async () => {
      await instance.login();
      expect(instance.props.reqLogin).toBeCalled();
    });
  });
  describe('Redux', () => {
    const initialState = {
      login: {
        data: {},
        error: false,
        isFetching: false,
      },
    };
    const store = mockStore(initialState);
    const component = shallow(<ComponentConnect store={store} />);

    it('mapStateToProps login should return true', () => {
      expect(component.props().login).not.toEqual({
        data: {},
        isFetching: false,
        error: false,
      });
    });
    it('mapDispatchToProps reqLogin call right type', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).reqLogin();
      expect(dispatch.mock.calls[0][0]).toEqual({
        payload: {
          username: undefined,
          password: undefined,
          isLogedIn: undefined,
        },
        type: 'REQ_LOGIN',
      });
    });
  });
});
