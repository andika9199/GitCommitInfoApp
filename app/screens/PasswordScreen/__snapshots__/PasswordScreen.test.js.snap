// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Screen Password Screen rendered Screen Renderer Successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/gitIcons.png",
      }
    }
    style={
      Object {
        "height": 270,
        "resizeMode": "contain",
        "width": 270,
      }
    }
  />
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "paddingHorizontal": 64,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 56,
          "paddingBottom": 16,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
            },
            Object {
              "left": 8,
              "right": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "bottom": -4,
                "left": -4,
                "position": "absolute",
                "right": -4,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "width": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": -4,
              "left": -4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 4,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": 0.5,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": -4,
            }
          }
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": 4,
            },
            Object {
              "width": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": -4,
              "left": 0,
              "position": "absolute",
              "right": -4,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "left": "-100%",
              "paddingLeft": "50%",
              "position": "absolute",
              "top": 0,
              "transform": Array [
                Object {
                  "scale": 1,
                },
                Object {
                  "translateY": 20,
                },
                Object {
                  "translateX": 0,
                },
              ],
              "width": "200%",
            }
          }
        >
          <Text
            numberOfLines={1}
            onTextLayout={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.38)",
                "fontSize": 16,
                "includeFontPadding": false,
                "lineHeight": 16,
                "textAlign": "left",
                "textAlignVertical": "top",
              }
            }
          >
            Password
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            disableFullscreenUI={true}
            editable={true}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="rgb(0, 145, 234)"
            style={
              Array [
                Object {
                  "flex": 1,
                  "includeFontPadding": false,
                  "margin": 0,
                  "padding": 0,
                  "paddingTop": 0,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                  "top": 2,
                },
                Object {
                  "color": "rgba(0, 0, 0, .87)",
                  "fontSize": 16,
                  "height": 24,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minHeight": 8,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#cccccc",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Robot",
          "fontSize": 13,
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
